### PR TITLE
Registration login #62

### LINKED ISSUE #62: Login/Registration Mask - Test with new frontend
**NOTE: This task depends on:**
- [x] Task #72 
- [ ] Task #73 (is not fully dependent. Consider to ONLY implement the described aspects below)

After the new frontend has been set-up, the Login/Register component should be tested.
- Implement Register form/mask and add button to the Login form/mask
- Implement basic features for the Authentication in the frontend
  - Login Page/Register should be displayed first when starting app
  - Implement Guard to not be able to navigate to other pages when not logged in (logged-in users receive a token from the backend)
  - consider using a npm module for handling the token and putting it automatically in the header for every request (i.e. https://github.com/auth0/angular2-jwt)
- Evaluate implemented features
  - Test Login
  - Test Register
  - Test Guards
  - Test JWT token handler

### PR DESCRIPTION
**Pull request by deKilla**
RESOLVES #62 

added:
- imported auth components from nebular (registration, login, logout)
- customized to our needs (incl. validation, roles etc.)
- fully tested implementation of register and login (works with the current implementation of the backend - subject to changes)

### CODE REVIEW

**Comment by deKilla** [comment]
please let me know that it is ok pre-merge (Telegram) as I'm going to add the Changelog at the earliest in the evening - thank you

**Comment by amarbajric** [comment]
**DONE:**
- checked register ✅ 
- checked login ✅ 
- checked async email validator ✅ 
- `CHANGELOG.md` missing 🔻 

**ToDo:**
- add entry in `CHANGELOG.md`
  - please add an extensive entry, no keywords or one-liner
Other than that the PR is **APPROVED**

**Review by amarbajric — CHANGES_REQUESTED** [review]
**DONE:**
- checked register ✅ 
- checked login ✅ 
- checked async email validator ✅ 
- `CHANGELOG.md` missing 🔻 

**ToDo:**
- add entry in `CHANGELOG.md`
  - please add an extensive entry, no keywords or one-liner

Other than that the PR is **APPROVED**

**Review by amarbajric — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,22 @@ This file ONLY contains changes made by the AIM17 dev-team. This means, that thi
 of the forked repository.
 The format is partially based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 
+## 2018-05-02
```
Comment (amarbajric) [inline CHANGELOG.md]: Link to PR is missing.
Please add

**Review by amarbajric — CHANGES_REQUESTED** [review]

**Review by amarbajric — APPROVED** [review]

**Pull request merged**